src/Fixer/Alias/ModernizeStrposFixer.php

```diff
@@ -0,0 +1,220 @@
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Alias;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\FixerDefinition\FixerDefinitionInterface;
+use PhpCsFixer\FixerDefinition\VersionSpecification;
+use PhpCsFixer\FixerDefinition\VersionSpecificCodeSample;
+use PhpCsFixer\Tokenizer\Analyzer\ArgumentsAnalyzer;
+use PhpCsFixer\Tokenizer\Analyzer\FunctionsAnalyzer;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author Alexander M. Turek <me@derrabus.de>
+ */
+final class ModernizeStrposFixer extends AbstractFixer
+{
+    private const REPLACEMENTS = [
+        [
+            'operator' => [T_IS_IDENTICAL, '==='],
+            'operand' => [T_LNUMBER, '0'],
+            'replacement' => [T_STRING, 'str_starts_with'],
+            'negate' => false,
+        ],
+        [
+            'operator' => [T_IS_NOT_IDENTICAL, '!=='],
+            'operand' => [T_LNUMBER, '0'],
+            'replacement' => [T_STRING, 'str_starts_with'],
+            'negate' => true,
+        ],
+        [
+            'operator' => [T_IS_NOT_IDENTICAL, '!=='],
+            'operand' => [T_STRING, 'false'],
+            'replacement' => [T_STRING, 'str_contains'],
+            'negate' => false,
+        ],
+        [
+            'operator' => [T_IS_IDENTICAL, '==='],
+            'operand' => [T_STRING, 'false'],
+            'replacement' => [T_STRING, 'str_contains'],
+            'negate' => true,
+        ],
+    ];
+
+    public function getDefinition(): FixerDefinitionInterface
+    {
+        return new FixerDefinition(
+            'Replace `strpos()` calls with `str_starts_with()` or `str_contains()` if possible.',
+            [
+                new VersionSpecificCodeSample(
+                    '<?php
+if (strpos($haystack, $needle) === 0) {}
+if (strpos($haystack, $needle) !== 0) {}
+if (strpos($haystack, $needle) !== false) {}
+if (strpos($haystack, $needle) === false) {}
+',
+                    new VersionSpecification(80000)
+                ),
+            ],
+            null,
+            'Risky if `strpos`, `str_starts_with` or `str_contains` functions are overridden.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Must run before BinaryOperatorSpacesFixer, NoExtraBlankLinesFixer, NoSpacesInsideParenthesisFixer, NoTrailingWhitespaceFixer, NotOperatorWithSpaceFixer, NotOperatorWithSuccessorSpaceFixer, SingleSpaceAfterConstructFixer.
+     */
+    public function getPriority(): int
+    {
+        return 37;
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return \PHP_VERSION_ID >= 80000 && $tokens->isTokenKindFound(T_STRING) && $tokens->isAnyTokenKindsFound([T_IS_IDENTICAL, T_IS_NOT_IDENTICAL]);
```

Comment:
In my original PR, I decided not to limit the fixer to PHP 8 because the new functions can be used on earlier PHP versions if you polyfill them (e.g. with `symfony/polyfill-php80`).

Reply:
interesting.. we could also check if the functions are defined in the global namespace at runtime.. I've no real strong preference, any of the other members have thoughts on this ? @keradus @julienfalque @kubawerlos

Reply:
tbh, I would not add any of those checks and assume that people know what they're doing if they opt-in to that rule.

Reply:
Checking whether the function exists won't work if the polyfill does not share the same autoloader as PHP CS Fixer (e.g. PHP CS Fixer installed globally or via a different composer.json file).

I would tend to agree with @derrabus but I'm not sure that would be consistent with how existing rules work, so I'll wait for more opinions to make my mind :)

Reply:
It would be consistent with `RandomApiMigrationFixer` which solves a similar issue.